cmd/semaphore/generate/protobuf/protobuf.go

```diff
@@ -0,0 +1,126 @@
+package protobuf
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/jexia/semaphore"
+	"github.com/jexia/semaphore/cmd/semaphore/config"
+	"github.com/jexia/semaphore/pkg/broker"
+	"github.com/jexia/semaphore/pkg/broker/endpoints"
+	"github.com/jexia/semaphore/pkg/broker/logger"
+	"github.com/jexia/semaphore/pkg/broker/providers"
+	"github.com/jexia/semaphore/pkg/codec/proto"
+	"github.com/jexia/semaphore/pkg/functions"
+	"github.com/jexia/semaphore/pkg/transport"
+	"github.com/jexia/semaphore/pkg/transport/grpc"
+	"github.com/jhump/protoreflect/desc/protoprint"
+	"github.com/spf13/cobra"
+)
+
+var params = config.New()
+
+// Command represents the semaphore daemon command
+var Command = &cobra.Command{
+	Use:   "protobuf",
+	Short: "Generates a protobuf definitions",
+	Long: `Generates a protobuf definitions.
+*NOTE*: This is a experimental feature and not all features are supported yet.`,
+	RunE:         run,
+	SilenceUsage: true,
+}
+
+func init() {
+	Command.PersistentFlags().StringSliceVar(&params.Protobuffers, "proto", []string{}, "If set are all proto definitions found inside the given path passed as schema definitions, all proto definitions are also passed as imports")
+	Command.PersistentFlags().StringSliceVarP(&params.Files, "file", "f", []string{"config.hcl"}, "Parses the given file as a definition file")
+	Command.PersistentFlags().StringVar(&params.LogLevel, "level", "warn", "Global logging level, this value will override the defined log level inside the file definitions")
+}
+
+func run(cmd *cobra.Command, args []string) error {
+	arguments, err := config.ConstructArguments(params)
+	if err != nil {
+		return err
+	}
+
+	ctx := logger.WithLogger(broker.NewContext())
+	options, err := semaphore.NewOptions(ctx, arguments...)
+	if err != nil {
+		return err
+	}
+
+	collection, err := providers.Resolve(ctx, functions.Collection{}, options)
+	if err != nil {
+		return err
+	}
+
+	transporters, err := endpoints.Transporters(ctx, collection.EndpointList, collection.FlowListInterface,
```

Comment:
I think you should be able to pass the EndpointList as well. The `endpoints.Transporters` method is used to initialize the transporters to be passed to the listeners.

Reply:
I ll check

Reply:
nope :disappointed:  
does not work, `grpc.ParseEndpointOptions` expects `*transport.Endpoint` to be passed